cmd/juju/application/deploy.go

```diff
@@ -1685,6 +1690,29 @@ func (c *DeployCommand) charmStoreCharm() (deployFn, error) {
 	}, nil
 }
 
+func deduceOrigin(url *charm.URL) application.CharmOrigin {
+	if url == nil {
+		return application.CharmOrigin{
+			Source: application.OriginUnknown,
+		}
+	}
+
+	switch url.Schema {
+	case "cs":
+		return application.CharmOrigin{
+			Source: application.OriginCharmStore,
+		}
+	case "local":
+		return application.CharmOrigin{
+			Source: application.OriginLocal,
+		}
+	default:
```

Comment:
Where does application.OriginCharmHub come into play here?  Or are we waiting for charm.URL to be updated with charmhub?  My understanding was that for deploy, you must specify "cs:" for charm store, a relative path name is a local charm and everything else is a charm hub charm.  Though I know of no specifier for a charm hub charm.

Reply:
Not sure yet, I had not wired that in yet. This was to ensure that the charm origin was wired up correctly to the db with the migrations.